### ISSUE TITLE
Auto-sync: Turn transcript text into captions like YouTube Studio and Descript

### ISSUE THREAD

**Issue by candideu** [Makememo/MemoAI#14]
**Describe the solution you'd like**
YouTube allows people to paste or upload a plain text transcript for a video, and it will automatically assign timings for the text to generate timed captions, which can then be exported as .srt, .vtt, etc. This feature is called [Auto-Sync](https://support.google.com/youtube/answer/2734796?sjid=7686619959639867344-NA#create&zippy=%2Cauto-sync). Descript offers a [similar syncing feature](https://www.youtube.com/watch?v=dzAexFlyNNI).

Would Whisper allow to create a similar feature in Memo?

**Comment by Makememo** [comment]
I understand that the procedure is to convert a video to text and then paste the text in Memo. Does Memo match the text with the subtitles? 

Am I getting this right? If so, we will support later versions.

**Comment by candideu** [comment]
Not quite. The procedure would be that user has a video or audio with text transcript of the media. The text has no timing information, and the user would like to generate captions/subtitles from that text. So the user provider the media and the text, and the platform automatically adds timing and syncs up the text to the media.

This tutorial provides a good demo of the feature on Descript: https://www.youtube.com/watch?v=dzAexFlyNNI

**Comment by Makememo** [comment]
Got It. Thank you!

**Comment by niubility000** [comment]
就是剪映里的文稿匹配功能，非常牛叉。你有一段视频，还有对应的一句一行的文稿，但没有时间轴信息，它可以给文本自动打轴，就是自动加上时间戳。剪映加的非常准，完全能给复读机单句复读用，就是文稿有5000字的限制。剪映用的不是whisper模型。据说whisper模型的时间戳都做不准，天生残疾。但我后来发现那个faster whisper medium 模型的时间戳基本非常准确（比现在memo ai用的准确），可以参考卡卡字幕助手videocaptioner 生成的词级的时间戳字幕，拿用户给出的文稿和这个词级字幕比对（应该是全文比对，先确定位置），再一次得到每句起止时间。但单词万一匹配不上呢？ 就报错呗。剪映的要求也挺严，视频比文稿长了，短了，或者瞎给文稿它都会乱匹配。

**Comment by candideu** [comment]
Storyteller is another app that does something similar, but for ebooks and audiobooks:

https://smoores.gitlab.io/storyteller/
